### PR TITLE
Add additional annotations to JobRunLeased event

### PR DESCRIPTION
**Pull request by msumner91**
Goal(s):
- Add AdditionalAnnotations to JobRunLeased events
- Expose gang_runtime_cardinality as an AdditionalAnnotation on JobRunLeased

A follow-up PR will:
- Consume these values & persist them into PG
- Add them as annotations on the resulting PodSpec

### CODE REVIEW

File: pkg/api/api.swagger.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ package api
 // SwaggerJsonTemplate is a generated function returning the template as a string.
 // That string should be parsed by the functions of the golang's template package.
 func SwaggerJsonTemplate() string {
-	var tmpl = "{\n" +
+	tmpl := "{\n" +
```
Comment (zuqq) [inline pkg/api/api.swagger.go:12]: You'll have to revert this change (and the analogous one below), or you'll break the tests; see #3084.

File: pkg/armadaevents/events.proto [hunk 2]
```diff
@@ -290,6 +290,8 @@ message JobRunLeased {
     string node_id = 4;
     // Used by the scheduler to maintain a consistent state
     int32 update_sequence_number = 5;
+    // Additional labels to be added to the PodSpec.
+    map<string, string> additional_labels = 6;
```
Comment (zuqq) [inline pkg/armadaevents/events.proto:294]: Do we really want to use labels instead of annotations?

Comment (msumner91) [reply pkg/armadaevents/events.proto:294]: I don't have a preference really - if an annotation makes more sense happy to use that

File: internal/scheduler/gang_scheduler.go [hunk 3]
```diff
@@ -204,18 +204,18 @@ func (sch *GangScheduler) trySchedule(ctx *armadacontext.Context, gctx *schedule
 	return sch.tryScheduleGang(ctx, gctx)
 }
 
-func (sch *GangScheduler) tryScheduleGang(ctx *armadacontext.Context, gctx *schedulercontext.GangSchedulingContext) (ok bool, unschedulableReason string, err error) {
+func (sch *GangScheduler) tryScheduleGang(ctx *armadacontext.Context, gctx *schedulercontext.GangSchedulingContext) (ok bool, runtimeGangCardinality int, unschedulableReason string, err error) {
```
[the remark this answers is not in the document]
Comment (msumner91) [reply internal/scheduler/gang_scheduler.go:207]: Ah possibly - that didn't exist when I last looked at this stuff.

Will take a look and see

File: pkg/armadaevents/events.proto (hunk 2 above)
Comment (msumner91) [reply pkg/armadaevents/events.proto:294]: The only reason we went for a label is is because we currently use those for similar things like job id / jobset / queue id.

File: internal/scheduler/gang_scheduler.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (msumner91) [reply internal/scheduler/gang_scheduler.go]: Changed this to use GangFit instead as it's cleaner to compute from the context -- thanks for the heads up

File: pkg/armadaevents/events.proto (hunk 2 above)
Comment (zuqq) [reply pkg/armadaevents/events.proto:294]: Looking at https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#attaching-metadata-to-objects, using an annotation might be more appropriate. Do we currently set any gang scheduling metadata on gang jobs (e.g., gang ID would arguably fall into the label category)?

Comment (msumner91) [reply pkg/armadaevents/events.proto:294]: Cool - I'll rename this to annotations then as it's not really something we want to select/identify with.

DogeFarm sets GangId etc. as metadata in DogeFarm and as an annotation to Armada

E.g. Yaml spec:
annotations:
      armadaproject.io/gangCardinality: '4'
      armadaproject.io/gangId: 28cf33a61192490a87974b0ad6eeb698-7

**Review by JamesMurkin — APPROVED** [review]
